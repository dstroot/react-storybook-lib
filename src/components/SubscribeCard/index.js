import React from 'react';
import MailchimpSubscribe from 'react-mailchimp-subscribe';
//https://www.npmjs.com/package/react-mailchimp-subscribe

// styles
import './styles.scss';

const url = process.env.REACT_APP_MAILCHIMP_URL;

const CustomForm = ({ status, message, onValidated }) => {
  let email;

  const submit = e => {
    email &&
      email.value.indexOf('@') > -1 &&
      onValidated({
        EMAIL: email.value,
      });
    e.preventDefault();
  };

  return (
    <div className="px-3 py-4 mb-3">
      <form onSubmit={submit}>
        <div className="input-group">
          <input
            ref={node => (email = node)}
            type="email"
            placeholder="Your email"
            className="form-control"
            value={status === 'success' ? '' : null}
          />
          <div className="input-group-append">
            <button type="submit" className="btn btn-secondary">
              Sign me up
            </button>
          </div>
        </div>
      </form>

      {status === 'sending' && <div style={{ color: 'blue' }}>sending...</div>}
      {status === 'error' && (
        <div
          class="alert alert-danger mb-0 mt-3 small"
          role="alert"
          dangerouslySetInnerHTML={{ __html: message }}
        />
      )}
      {status === 'success' && (
        <div
          class="alert alert-success mb-0 mt-3 small"
          role="alert"
          dangerouslySetInnerHTML={{ __html: message }}
        />
      )}
    </div>
  );
};

const SubscribeMailchimp = () => {
  return (
    <MailchimpSubscribe
      url={url}
      render={({ subscribe, status, message }) => (
        <CustomForm
          status={status}
          message={message}
          onValidated={formData => subscribe(formData)}
        />
      )}
    />
  );
};

const SubscribeCard = () => (
  <div className="container-fluid bg-light">
    <div className="row">
      <div className="col-12">
        <div className="m-5 align-items-center bg-white p-3">
          <h5 className="text-muted">Get hooked up!</h5>
          <p className="lead">Enter your email to get weekly updates.</p>
          <SubscribeMailchimp />
        </div>
      </div>
    </div>
  </div>
);

export default SubscribeCard;
