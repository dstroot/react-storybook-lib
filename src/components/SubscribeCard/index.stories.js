import React from 'react';
import { storiesOf } from '@storybook/react';
// import { Meta, Story, Props } from '@storybook/addon-docs/blocks';
// import { Meta } from '@storybook/addon-docs/blocks';

// components
import SubscribeCard from '../SubscribeCard';

// SubscribeCard section
storiesOf('Pac Life|SubscribeCard', module)
  .addParameters({
    info: {
      inline: true,
      header: false,
    },
  })
  .add('show SubscribeCard', () => <SubscribeCard />);
