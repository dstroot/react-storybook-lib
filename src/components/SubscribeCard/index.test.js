import React from 'react';
import renderer from 'react-test-renderer';

import SubscribeCard from '../SubscribeCard';

describe('SubscribeCard', () => {
  it('it should render', () => {
    const component = renderer.create(<SubscribeCard />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
