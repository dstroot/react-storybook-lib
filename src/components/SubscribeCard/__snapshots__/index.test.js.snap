// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubscribeCard it should render 1`] = `
<div
  className="container-fluid bg-light"
>
  <div
    className="row"
  >
    <div
      className="col-12"
    >
      <div
        className="m-5 align-items-center bg-white p-3"
      >
        <h5
          className="text-muted"
        >
          Get hooked up!
        </h5>
        <p
          className="lead"
        >
          Enter your email to get weekly updates.
        </p>
        <div
          className="px-3 py-4 mb-3"
        >
          <form
            onSubmit={[Function]}
          >
            <div
              className="input-group"
            >
              <input
                className="form-control"
                placeholder="Your email"
                type="email"
                value={null}
              />
              <div
                className="input-group-append"
              >
                <button
                  className="btn btn-secondary"
                  type="submit"
                >
                  Sign me up
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
